packages/core/admin/admin/src/content-manager/components/DynamicTable/index.js

```diff
@@ -95,6 +95,7 @@ const DynamicTable = ({
         rows={rows}
         withBulkActions
         withMainAction={canDelete && isBulkable}
+        layout={layout}
```

Comment:
You don't need to pass the layout here but what you can do is to change the mainField in the component's layout

```
{
   metadatas: { //..., mainField: { type, name } 
}
```

So you have all the info needed in your RepeatableCell.